fastlane_core/lib/fastlane_core/queue_worker.rb

```diff
@@ -6,7 +6,7 @@ module FastlaneCore
   # Use this when you have all the items that you'll process in advance.
   # Simply enqueue them to this and call `QueueWorker#start`.
   class QueueWorker
-    NUMBER_OF_THREADS = FastlaneCore::Helper.test? ? 1 : [(ENV["DELIVER_NUMBER_OF_THREADS"] || ENV.fetch("FL_NUMBER_OF_THREADS", 10)).to_i, 10].min
+    NUMBER_OF_THREADS = FastlaneCore::Helper.test? ? 1 : [(ENV["DELIVER_NUMBER_OF_THREADS"] || ENV.fetch("FL_NUMBER_OF_THREADS", 10)).to_i, 1].max
```

Comment:
I was confused as well about what was the intention of this line and how we set the number of threads 😅 , so I initially  thought it would be best to do a change that's close to the original behaviour.

Capping the value of the max number of threads spawned by fastlane sounds like a sensible choice and I agree we should do it (_we don't know what users might do later_ 😄). But I would also need a way to "unlock" that maximum cap for the reasons I stated above (_which mostly applies to users who understand the consequence_), so how about adding another env var called `FL_MAX_NUMBER_OF_THREADS` with a default value of `10`? Improving on your suggestion, it would look like so:

```suggestion
    NUMBER_OF_THREADS = FastlaneCore::Helper.test? ? 1 : [ENV["DELIVER_NUMBER_OF_THREADS"], ENV["FL_NUMBER_OF_THREADS"], 10].map(&:to_i).find(&:positive?).clamp(1, ENV.fetch("FL_MAX_NUMBER_OF_THREADS", 10).to_i)
```

Reply:
Yeah sounds like a great balance.